firmware/hw_layer/mc33816.cpp

```diff
@@ -117,6 +110,51 @@ static void setup_spi() {
 	spiUnselect(driver);
 }
 
+static unsigned short readId() {
+	spiSelect(driver);
+	spi_writew(0xBAA1);
+	unsigned short ID =  recv_16bit_spi();
+	spiUnselect(driver);
+	return ID;
+}
+
+// Read a single word in Data RAM
+unsigned short mcReadDram(MC33816Mem addr) {
+	unsigned short readValue;
+	spiSelect(driver);
+	// Select Channel command, Common Page
+    spi_writew(0x7FE1);
+    spi_writew(0x0004);
+    // read (MSB=1) at data ram x9 (SCV_I_Hold), and 1 word
+    spi_writew((0x8000 | addr << 5) + 1);
+    readValue = recv_16bit_spi();
+
+    spiUnselect(driver);
+    return readValue;
+}
+
+// Update a single word in Data RAM
+void mcUpdateDram(MC33816Mem addr, unsigned short data) {
+	spiSelect(driver);
+	// Select Channel command, Common Page
+    spi_writew(0x7FE1);
+    spi_writew(0x0004);
+    // write (MSB=0) at data ram x9 (SCV_I_Hold), and 1 word
+    spi_writew((addr << 5) + 1);
+    spi_writew(data);
+
+    spiUnselect(driver);
+}
+
+void setBoostVoltage(float volts)
+{
+	// Specifically for Discovery's ~4.5v 5v rail
+	unsigned short data = volts * 3.2;
```

Comment:
This isn't an artifact, the correct value from the datasheet is 3.2 counts per volt.

There's a 1/32 divider on the input, then the DAC's output is 9.77mV per LSB.  (1 / 32) / 0.00977 = 3.199 counts per volt.